test/unit/math/prim/mat/vectorize/expect_std_vector_error.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_EXPECT_STD_VECTOR_ERROR_HPP
+#define TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_EXPECT_STD_VECTOR_ERROR_HPP
+
+#include <gtest/gtest.h>
+#include <stdexcept>
+#include <vector>
+
+template <typename F, typename T>
+void expect_std_vector_error() {
+  using std::vector;
+  vector<double> invalid_inputs = F::invalid_inputs();
+  if (invalid_inputs.size() > 0) {
```

Comment:
Stylistically, I'd prefer a test right at the top of:

```
if (F::invalid_inputs().size() == 0) return;
```

This achieves two desirable goals.  One, it makes it clear that you're checking boundary conditions and dealing with them separately before the main code.  Two, all the rest of the code is less nested, which makes everything much easier to read.  (I'd strongly recommend Martin Fowler's book <i>Refactoring</i>, by the way.)

Anyway, it's no big deal here in the tests, just a heads up.